crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -146,6 +146,16 @@ pub fn collect_function_level_profiling_info(
     ];
 
     let sierra_statements = map_pcs_to_sierra_statement_ids(casm_debug_info, casm_level_info);
+    // get all sizes to a hashmap, for quicker lookup
```

Comment:
#166 
this is rather a workaround than a solid, final solution. The thing is I will need to make further changes to function trace builder and cairo-annotations, most probably.